pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"net/http"
+
+	apimodels "github.com/grafana/alerting-api/pkg/api"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/state"
+)
+
+type PrometheusSrv struct {
+	log          log.Logger
+	stateTracker *state.StateTracker
+}
+
+func (srv PrometheusSrv) RouteGetAlertStatuses(c *models.ReqContext) response.Response {
+	alertResponse := apimodels.AlertResponse{
+		DiscoveryBase: apimodels.DiscoveryBase{
+			Status: "success",
+		},
+		Data: apimodels.AlertDiscovery{
+			Alerts: []*apimodels.Alert{},
+		},
+	}
+	for _, alertState := range srv.stateTracker.GetAll() {
+		startsAt := alertState.StartsAt
+		alertResponse.Data.Alerts = append(alertResponse.Data.Alerts, &apimodels.Alert{
+			Labels:      map[string]string(alertState.Labels),
+			Annotations: map[string]string{}, //TODO: Once annotations are added to the evaluation result, set them here
+			State:       alertState.State.String(),
+			ActiveAt:    &startsAt,
```

Comment:
I get `"activeAt": "0001-01-01T01:34:52+01:34"` for all my alerts; I wonder if it's something wrong with my data

Reply:
I think I see why. My alerts all had `0000-12-31T16:07:02-07:52` for their time stamp. There's more work I need to do around handling things that start reporting in an alerting state, for the first time. In this case, `getOrCreate` was not setting the `StartsAt` field. My thought, at the time, was that we couldn't know when it went into that state but it makes more sense to set it to be `result.EvaluatedAt` in this case since that is when this AlertState went into `alerting` small fix incoming for that. Nice catch @papagian , thank you!